sync3/handler/connstate_live.go

```diff
@@ -259,7 +259,13 @@ func (s *connStateLive) processLiveUpdate(ctx context.Context, up caches.Update,
 			if delta.RoomNameChanged {
 				metadata := roomUpdate.GlobalRoomMetadata()
 				metadata.RemoveHero(s.userID)
-				thisRoom.Name = internal.CalculateRoomName(metadata, 5) // TODO: customisable?
+				roomName, calculated := internal.CalculateRoomName(metadata, 5) // TODO: customisable?
+
+				thisRoom.Name = roomName
+
+				if s.roomSubscriptions[roomUpdate.RoomID()].IncludeHeroes() && calculated {
```

Comment:
This isn't enough as it won't work for lists. We need to instead find out which subscriptions this room relates to and use the values from that. Call `roomIDsToLists := s.lists.ListsByVisibleRoomIDs(s.muxedReq.Lists)` and see if any of those list keys have heroes included in `s.muxedReq`.